src/himawari_ahi_mod.F90

```diff
@@ -0,0 +1,1666 @@
+        phase_file = fnames(ifile)
+     else if ( is_HT(ifile) ) then
+        got_ht = .true.
+        ht_file = fnames(ifile)
+     end if
+
+     valid(ifile) = .true.
+
+     ! group files of the same scan time
+     if ( t_index == 0 ) then
+        t_index = t_index + 1
+        ftime_id(ifile) = t_index
+     else
+        found_time = .false.
+        find_time_loop: do ii = ifile-1, 1, -1
+           if ( valid(ii) ) then
+              if ( scan_time(ifile) == scan_time(ii) ) then
+                 ftime_id(ifile) = ftime_id(ii)
+                 found_time = .true.
+                 exit find_time_loop
+              end if
+           end if
+        end do find_time_loop
+        if ( .not. found_time ) then
+           t_index = t_index + 1
+           ftime_id(ifile) = t_index
+        end if
+     end if
+     ntime = t_index
+
+   end do file_loop1
+
+   if ( ntime <= 0 ) then
+      write(0,*) 'ntime = ', ntime
+      write(0,*) 'No valid files found from hs_list_file '//trim(hs_list_file)
+      stop
+   else if ( ntime > 1 ) then
+      write(0,*) 'ntime = ', ntime
+      write(0,*) 'Multiple time file NOT supported yet'
+      stop
+   else
+      it = ftime_id(1)
+   end if
+
+   allocate (time_start(ntime))
+   allocate (rdata(ntime))
+
+   if ( any(valid) ) then
+      if ( got_hs ) then
+
+         allocate (got_latlon_out(npixel, nline))
+         allocate (glat_out(npixel, nline))
+         allocate (glon_out(npixel, nline))
+         allocate (gsolzen(npixel, nline))
+         allocate (gsatzen(npixel, nline))
+         glat_out(:,:)  = missing_r
+         glon_out(:,:)  = missing_r
+         gsolzen(:,:)   = missing_r
+         gsatzen(:,:)   = missing_r
+
+         allocate (F_out(npixel, nline, nband+1)) !IHB BJJ: 1:nband for bt, nband+1 for # of obs for SO
+         F_out(:,:,:)   = missing_r
+
+         allocate (varname_out(nband+1))          !IHB BJJ: 1:nband for bt, nband+1 for # of obs for SO
+         varname_out(:) = ''
+
+         nsegm = segm(1)
+         call read_HS_HSD(data_dir, fnames, fsat_id, nsegm, julianday, glon_out, glat_out, brit, gsolzen, gsatzen, got_latlon_out)
+
+         if ( .not. allocated(rdata(it)%rad) ) allocate (rdata(it)%rad(nband,npixel, nline))
+         if ( .not. allocated(rdata(it)%bt) )  allocate (rdata(it)%bt(nband,npixel, nline))
+         if ( .not. allocated(rdata(it)%qf) )  allocate (rdata(it)%qf(nband,npixel, nline))
+         if ( .not. allocated(rdata(it)%sd) )  allocate (rdata(it)%sd(nband))
+
+         do ib = 1, nband
+            do j = 1, nline
+               do i = 1, npixel
+                  rdata(it)%rad(ib, i, j) = missing_r
+                  rdata(it)%bt(ib, i, j) = brit(i, j, ib)
+                  rdata(it)%qf(ib, i, j) = missing_r
+                  rdata(it)%sd(ib) = missing_r
+                  F_out(i, j, ib) = brit(i, j, ib)
+
+               end do
+            end do
+            write(varname_out(ib),"(A,I2.2)") 'BT_'//fsat_id//'C', ib
```

Comment:
`ib` ranges from `1` to `nband` here, but the typical channel ranges from `7` to `16`. 

Can we use `fband_id` variable here? or we can even set `ib+6` with some assumption. The latter approach actually is being used in `subroutine output_iodav1` and `subroutine output_iodav1_o2m`.

Reply:
This is a good point! I modified it in my latest commit. Thanks @byoung-joo!